packages/wdio-devtools-service/src/index.js

```diff
@@ -29,23 +36,133 @@ export default class DevToolsService {
 
         try {
             let debuggerAddress
+
             if (this.options.debuggerAddress) {
                 const [host, port] = this.options.debuggerAddress.split(':')
                 debuggerAddress = { host, port: parseInt(port, 10) }
             } else {
                 debuggerAddress = await findCDPInterface()
             }
-            const client = await getCDPClient(debuggerAddress)
-            this.commandHandler = new CommandHandler(client, global.browser)
+
+            this.client = await getCDPClient(debuggerAddress)
+            this.commandHandler = new CommandHandler(this.client, global.browser)
+            this.devtoolsDriver = await DevToolsDriver.attach(`http://${debuggerAddress.host}:${debuggerAddress.port}`)
+            this.traceGatherer = new TraceGatherer(this.devtoolsDriver)
+
+            const session = await this.devtoolsDriver.getCDPSession()
+            session.on('Page.loadEventFired', ::this.traceGatherer.onLoadEventFired)
+            session.on('Page.frameNavigated', ::this.traceGatherer.onFrameNavigated)
+
+            const page = await this.devtoolsDriver.getActivePage()
+            page.on('requestfailed', ::this.traceGatherer.onFrameLoadFail)
+
+            /**
+             * enable domains for client
+             */
+            await Promise.all(['Page', 'Network', 'Console'].map(
+                (domain) => Promise.all([
```

Comment:
it might be clearer to avoid second `Promise.all()`. It would look like:
```js
await Promise.all(['Page', 'Network', 'Console'].map(
   async (domain) => {
       await session.send(`${domain}.enable`)
       this.client[domain]['enable']()
    }
...
```

Reply:
This way, it would take twice as long because both of the commands aren't run in parallel.